projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1122,6 +1115,23 @@ function get_paywall_blocks( $newsletter_access_level ) {
 ';
 }
 
+/**
+ * Returns true if user is auth for subscriptions check, otherwise returns false.
+ *
+ * @return boolean
+ */
+function is_user_auth() {
+	if ( is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
+		return true;
+	}
+	if ( class_exists( 'Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Jetpack_Token_Subscription_Service' ) ) {
+		if ( Jetpack_Token_Subscription_Service::has_token_from_cookie() ) {
+			return true;
```

Comment:
This assumption is false: the user can have the code and NOT be logged in to WPCOM.

For example: 
- When coming back from the checkout page, user has the subscription token but is NOT logged in to WPCOM.
- WPCOM session can expire or the user log out of WPCOM and still have the token on the hosted site

Reply:
@millerf  What's the consiquence? 

If they're seeing the subscriber-only content at the site because of the token is present, but not paid content because they're not paid member, we suggest them to switch accounts based on this check?

Reply:
`has_access_to_subscriber_only_content` would be very clear? :-)

Reply:
> we suggest them to switch accounts based on this check?

I agree this is not the best wording. The "Switch account" is here if they entered the wrong email for exmaple when "getting access". Otherwise they have no way to remove the "wrong" token...

Reply:
Use case is:
- User has a xx@gmail.com email address and has previously paid for the subscription and should have access
- User enters yx@gmail.com by mistake, or because they ix up their account
- Then user does not have a way to get access with xx@gmail.com, unless we have a "switch account" flow...

Reply:
> has_access_to_subscriber_only_content would be very clear? :-)

I kind of see what you are trying to do there, but:
- on WPCOM, you could be logged in WPCOM and not be a site subscriber
- on self-hosted, you could have a token and your subscription still pending (email not verified)

But I agree it means they have an account on WPCOM. So `has_wpcom_account()` would work...

Even after that, I don't see how does removing the "iframe flow" on self-hosted improves the user experience...

Reply:
> on WPCOM, you could be logged in WPCOM and not be a site subscriber

so you get a switch account link.

> on self-hosted, you could have a token and your subscription still pending (email not verified)

you will get the message in the paywall. to verify the email.